documentation/docs/30-advanced/20-hooks.md

```diff
@@ -61,7 +61,14 @@ export async function handle({ event, resolve }) {
 	event.locals.user = await getUserInformation(event.cookies.get('sessionid'));
 
 	const response = await resolve(event);
-	response.headers.set('x-custom-header', 'potato');
+
+	try {
+		response.headers.set('x-custom-header', 'potato');
+	} catch {
+		// Response objects can have immutable headers.
+		// Modifying immutable headers throws a TypeError.
+		// e.g. Response.redirect() returned from an endpoint.
+	}
```

Comment:
what about this?

```suggestion

	// Note that response objects can have immutable headers
	// (e.g. Response.redirect() returned from an endpoint).
	// Modifying immutable headers throws a TypeError.
	// In that case, clone the response or avoid creating a
	// response object with immutable headers.
	response.headers.set('x-custom-header', 'potato');
```

that way we note it but don't have to bloat a simple code snippets